src/core/src/engine.rs

```diff
@@ -295,13 +288,14 @@ impl Engine for Poet2Engine {
                             // wait_time = service.calculate_wait_time(new_chain_head.clone());
                             published_at_height = false;
                             start = time::Instant::now();
-                            unsafe {
-                                CLAIM_BLOCK_DUR = service.get_next_wait_time();
-                            }
-                            service.initialize_block(Some(new_chain_head));
+                            let chain_head_block = service.get_chain_head();
+                            wait_time = time::Duration::from_secs(service.get_wait_time(chain_head_block.clone(), &validator_id));
```

Comment:
Is the complete scope required here? time::Duration::from_secs??

Reply:
why not?

Reply:
use std::time::Duration;
^ This is already there. Why have a fully qualified name then?